l10n_it_pec_messages/model/fetchmail.py

```diff
@@ -41,4 +49,241 @@ class FetchmailServer(orm.Model):
         'out_server_id': fields.one2many(
             'ir.mail_server', 'in_server_id',
             'Outgoing Server',  readonly=True),
+        'last_internal_date': fields.datetime('Last Internal Date'),
     }
+
+    def _fetch_from_data_imap(self, cr, uid,
+                              server, imap_server,
+                              mail_thread, action_pool,
+                              count, failed,
+                              context=None):
+        messages = []
+        date_uids = {}
+        last_date = False
+        last_internal_date = datetime.strptime(
+            server.last_internal_date,
+            "%Y-%m-%d %H:%M:%S")
+        timestamp1 = calendar.timegm(
+            last_internal_date.timetuple())
+        intDate = imaplib.Time2Internaldate(timestamp1)
+        search_status, uids = imap_server.search(
+            None,
+            'SINCE',
+            '%s' % intDate
+            )
+        new_uids = uids[0].split()
+        for new_uid in new_uids:
+            fetch_status, data = imap_server.fetch(
+                int(new_uid),
+                'INTERNALDATE'
+                )
+            internaldate = imaplib.Internaldate2tuple(data[0])
+            internaldate_msg = datetime.fromtimestamp(
+                time.mktime(internaldate)
+                )
+            if internaldate_msg > last_internal_date:
+                messages.append(new_uid)
+                date_uids[new_uid] = internaldate_msg
+        for num in messages:
+            # SEARCH command *always* returns at least the most
+            # recent message, even if it has already been synced
+            res_id = None
+            result, data = imap_server.uid('fetch', num,
+                                           '(RFC822)')
+            imap_server.store(num, '-FLAGS', '\\Seen')
+            try:
+                res_id = mail_thread.message_process(
+                    cr, uid,
+                    server.object_id.model,
+                    data[0][1],
```

Comment:
@angedras  sono riuscito a farlo arrivare qui, mi trova una buona serie di messaggi in base alla data ma poi data[0] e' None quindi ho aggiunto un if prima del try.